src/components/ButtonGroup/GroupButton/index.tsx

```diff
@@ -1,11 +1,55 @@
 import React from 'react';
 import cc from 'classcat';
-import PropTypes from 'prop-types';
 import baseElement from 'util/baseElement';
+import Icon from 'components/Icon';
 
-const WrapperComponent = (wrapperProps) => wrapperProps.children;
+const WrapperComponent = ({ children }: { children: JSX.Element }): JSX.Element =>
+  children;
 
-export const GroupButton = (props) => {
+export interface GroupButtonProps {
+  /** Displays active button style when true */
+  isActive?: boolean;
+
+  /**
+   * Takes in a react-router `Link` reference and sets it
+   * as the base element. You may ONLY use it like the
+   * following:
+   *
+   * - `import { Link } from 'react-router'`
+   * - `Link={Link}`
+   */
+  Link?: JSX.Element;
```

Comment:
in other files we have this as `Link?: React.ElementType;` or `Link?: React.ComponentType`, we should probably decide on one and stick to that.

@reywright what do you think?

Reply:
@ivanhdez28 I agree, though I think my question before we standardize, is which one is correct?

Reply:
`JSX.Element` is a `ReactElement` whose props and type are `any`
```
declare global {
  namespace JSX {
    interface Element extends React.ReactElement<any, any> { }
  }
}
```

```
type ElementType<P = any> =
    {
        [K in keyof JSX.IntrinsicElements]: P extends JSX.IntrinsicElements[K] ? K : never
    }[keyof JSX.IntrinsicElements] |
    ComponentType<P>;
type ComponentType<P = {}> = ComponentClass<P> | FunctionComponent<P>;
```

i feel like `ComponentType` is enough but i need to do more research on this topic to make an educated comment.

Reply:
Given that this PR has been open for a week, I would bias moving forward with these changes, with the knowledge that I think as we proceed, answers around what we should standardize around, and in what situations, will be made a lot more clear (especially as we start converting files in `cbi-site` to TypeScript).